tests/basic_tests.cpp

```diff
@@ -462,4 +462,13 @@ TYPED_TEST(basic_tests, negativeport) {
   auto url = ada::parse<TypeParam>("https://www.google.com");
   ASSERT_FALSE(url->set_port("-1"));
   SUCCEED();
-}
\ No newline at end of file
+}
+
+// https://github.com/ada-url/ada/issues/826
+TYPED_TEST(basic_tests, set_invalid_port) {
+  auto url = ada::parse<TypeParam>("fake://dummy.test");
+  ASSERT_TRUE(url);
+  ASSERT_FALSE(url->set_port("invalid80"));
+  ASSERT_TRUE(url->is_valid);
```

Comment:
Let's make sure the port number did not change:

```suggestion
  ASSERT_FALSE(url->set_port("invalid80"));
  ASSERT_EQ(url->get_port(), "");
  ASSERT_TRUE(url->is_valid);
```